spark-launch/src/main/scala/com/ibm/sparktc/sparkbench/sparklaunch/ConfigWrangler.scala

```diff
@@ -32,23 +30,23 @@ object ConfigWrangler {
     * @param path
     * @return a Seq of paths to the created files.
     */
-  def apply(path: File): Seq[SparkLaunchConf] = {
+  def apply(path: File): Seq[(SparkLaunchConf, String)] = {
     val config: Config = ConfigFactory.parseFile(path)
     val sparkBenchConfig = config.getObject("spark-bench").toConfig
     val sparkContextConfs = getConfigListByName("spark-contexts", sparkBenchConfig)
 
-    val tmpDir = Files.createTempDirectory("spark-bench-tempfiles-")
-    val tmpFiles: Seq[Path] = sparkContextConfs.indices.map(_ => Files.createTempFile(tmpDir, "spark-bench-", ".conf"))
-
-    sparkContextConfs.indices.map( i =>
-    {
+    sparkContextConfs.map { conf =>
+      val tmpFile = Files.createTempFile("spark-bench-", ".conf")
       // Write this particular spark-context to a separate file on disk.
-      val output = sparkContextConfs(i).root().render(ConfigRenderOptions.defaults())
-      val str = s"spark-bench : {\nspark-contexts : [$output]\n}"
-      Files.write(tmpFiles(i), str.getBytes(), StandardOpenOption.CREATE, StandardOpenOption.TRUNCATE_EXISTING)
+      val newConf = sparkBenchConfig
+        .withoutPath("spark-contexts")
+        .withValue("spark-contexts", ConfigValueFactory.fromIterable(Iterable(conf.root).asJava))
+      val sbConf = ConfigFactory.empty.withValue("spark-bench", newConf.root)
+      val output = sbConf.root.render(concise.setJson(false))
+      Files.write(tmpFile, output.getBytes(), StandardOpenOption.CREATE, StandardOpenOption.TRUNCATE_EXISTING)
       // Now create the SparkLaunchConfs
-      SparkLaunchConf(sparkContextConfs(i), tmpFiles(i).toString)
-    })
+      (SparkLaunchConf(conf, tmpFile.toString), tmpFile.toString)
```

Comment:
This is a note to self about what we talked about IRL. This is kind of weird, but the reason for returning the tmpFile path separately even when it's included in the SparkLaunchConf case class is because that case class has an Array[String] for childArgs and you can't guarantee that childArgs.head will give you the right thing. So this is for future extensibillity.